core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1085,6 +1085,10 @@ class DAGScheduler(
 
         if (disallowStageRetryForTest) {
           abortStage(failedStage, "Fetch failure will not retry stage due to testing config")
+        } else if (failedStage.failAndShouldAbort()) {
+          abortStage(failedStage, s"Fetch failure - aborting stage. Stage ${failedStage.name} " +
```

Comment:
Can you remove the first sentence of this and also change "Failure reason" to say "Most recent failure reason:"